.github/workflows/main-cmake.yml

```diff
@@ -0,0 +1,140 @@
+name: NetCDF-C CMake CI - Windows
+
+on: [pull_request, workflow_dispatch]
+
+concurrency:  
+  group: ${{ github.workflow }}-${{ github.head_ref }}  
+  cancel-in-progress: true
+  
+jobs:
+
+  cmake_build_and_test:
+    strategy:
+      
+        matrix:
+            name: 
+                - "Windows MSVC"
+            hdf5:
+                - "1.14.3"
+
+            # Visual Studio + CMake
+            include:
+                - name: "Windows MSVC"
+                  os: windows-latest
+                  generator: "-G  \"Visual Studio 17 2022\""
+
+
+    name: "${{ matrix.name }}"
+
+    runs-on: ${{ matrix.os }}
+
+    # Each step in the job.
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    steps:
```

Comment:
What compilers do you want to target with this CI? You should specify the compiler used via `-DCMAKE_C_COMPILER` (or preferably presets) to make sure the appropriate compilers are used.

For MSVC compilers, you would probably have to add `uses: ilammy/msvc-dev-cmd@v1` and point the compiler to `cl`

Reply:
Currently this will just be used with the Visual Studio compiler; I need to get a basic test in for Windows + Visual Studio.  It will be expanded/revamped down the road, but I'd like to get this in for the v4.9.3 release we need to get out this week.  Code compilation is running fine right now, the runtime errors being reported resolve to 'missing DLL', which suggests a pathing issue at runtime.